poly/promise/promise.go

```diff
@@ -0,0 +1,744 @@
+package promise
+
+import (
+	"encoding/binary"
+	"errors"
+	"io"
+	"reflect"
+	"strconv"
+	"time"
+
+	"github.com/dedis/crypto/abstract"
+	"github.com/dedis/crypto/anon"
+	"github.com/dedis/crypto/config"
+	"github.com/dedis/crypto/poly"
+	"github.com/dedis/crypto/random"
+)
+
+// TODO Pass BlameProof as pointer. Consider generalizing it.
+// TODO Add Equal, Marshal, and UnMarshal methods for all
+// TODO Add tests for things I haven't yet.
+// TODO In tests, only use basicPromise if you ain't going to change it.
+// TODO Check the valdidity of PromiseSignature and BlameProof more extensively.
+//      make sure same suite, index proper, etc.
+// TODO Create valid promise to do basic sanity checking.
+// TODO Combine the valid* and Verify*
+// TODO Decouple keysuite from sharesuite
+// TODO It should be i >= p.n
+
+/* The PromiseSignature object is used for guardians to express their approval
+ * of a given promise. After receiving a promise and verifying that their share
+ * is good, guardians can then produce a signature to send back to the promiser.
+ *
+ * Upon receiving this, the promiser can then add the signature to its lists of
+ * signatures to server as proof that the promiser has gained a sufficient
+ * number of guardians.
+ */
+type PromiseSignature struct {
+
+	// The index of the guardian producing the signature
```

Comment:
I removed the index (which must now be tracked by the caller). In short, PromiseSignature is a simple struct that makes keeping track of signatures in BlameProofs and PromiseState easier. It handles the marshalling/ unmarshalling of the signature itself, which helps make other code simpler.